Applications/Utils/ModelPreparation/PartitionMesh/NodeWiseMeshPartitioner.h

```diff
@@ -0,0 +1,150 @@
+/*!
+  \file NodeWiseMeshPartitioner.h
+  \date   2016.05
+
+  \brief  Declare a class to perform node wise mesh partitioning
+
+  \copyright
+  Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+             Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+
+*/
+
+#ifndef NODE_WISE_MESH_PARTITIONER_H_
+#define NODE_WISE_MESH_PARTITIONER_H_
+
+#include <memory>
+#include <vector>
+#include <string>
+#include <fstream>
+
+#include "MeshLib/Mesh.h"
+
+namespace ApplicationUtils
+{
+///  A subdomain mesh.
+struct Partition
+{
+    std::vector<MeshLib::Node*> nodes;  ///< nodes.
+    std::size_t number_of_non_ghost_base_nodes;
+    std::size_t number_of_non_ghost_nodes;
+    std::size_t number_of_base_nodes;
+    /// Non ghost elements
+    std::vector<const MeshLib::Element*> regular_elements;
+    std::vector<const MeshLib::Element*> ghost_elements;
+};
+
+/// Mesh partitioner.
+class NodeWiseMeshPartitioner
+{
+public:
+    typedef long IntegerType;
+
+public:
+    /*!
+     * \param file_name_base The prefix of the file name.
+     * \param num_partitions Number of partitions,
+     * \param mesh           Pointer to a mesh object.
+     */
+    NodeWiseMeshPartitioner(const IntegerType num_partitions,
+                            std::unique_ptr<MeshLib::Mesh>& mesh)
+        : _npartitions(num_partitions),
+          _partitions(num_partitions),
+          _mesh(std::move(mesh)),
```

Comment:
move and non-rvalue references do not match. It certainly compiles, but I do not fully understand the operations here. I'd expect the `mesh` argument to be a rvalue reference.
Could you please explain the difference between
```c++
Partitioner(std::unique_ptr<Mesh>& mesh) : _mesh(std::move(mesh)) {}
```
and
```c++
Partitioner(std::unique_ptr<Mesh>&& mesh) : _mesh(std::move(mesh)) {}
```